artemis-protocols/artemis-mqtt-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/mqtt/MQTTProtocolManagerFactory.java

```diff
@@ -64,4 +74,46 @@ public ProtocolManager createProtocolManager(ActiveMQServer server,
    public String getModuleName() {
       return MODULE_NAME;
    }
+
+
+   @Override
+   public void loadProtocolServices(ActiveMQServer server, List<ActiveMQComponent> services) {
+      services.add(new MQTTPeriodicTasks(server.getScheduledPool()));
+   }
+
+   public class MQTTPeriodicTasks implements ActiveMQComponent {
+      ScheduledExecutorService pool;
+      Runnable runnable;
+
+      public MQTTPeriodicTasks(ScheduledExecutorService pool) {
+         this.pool = pool;
+      }
+
+      @Override
+      public void start() throws Exception {
+         runnable = () -> {
+            for (Map.Entry<String, MQTTSessionState> entry : sessionStates.entrySet()) {
+               MQTTSessionState state = entry.getValue();
+               MQTTLogger.LOGGER.debug("Inspecting session state: " + state);
```

Comment:
We are not using this type of logger any more for debug


just get an individual logger on the class you're at...

and just use logger.debug(....)